x/pairing/types/relay_finalization.go

```diff
@@ -0,0 +1,33 @@
+package types
+
+import (
+	"bytes"
+	"encoding/binary"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tendermintcrypto "github.com/tendermint/tendermint/crypto"
+)
+
+type RelayFinalization struct {
+	Exchange RelayExchange
+	Addr     sdk.AccAddress
+}
+
+func NewRelayFinalization(exch RelayExchange, addr sdk.AccAddress) RelayFinalization {
+	return RelayFinalization{Exchange: exch, Addr: addr}
+}
+
+func (rf RelayFinalization) GetSignature() []byte {
+	return rf.Exchange.GetSignature()
+}
+
+func (rf RelayFinalization) PrepareForSignature() []byte {
+	relaySessionHash := tendermintcrypto.Sha256(rf.Exchange.Request.RelaySession.CalculateHashForFinalization())
+	latestBlockBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(latestBlockBytes, uint64(rf.Exchange.Reply.LatestBlock))
+	return bytes.Join([][]byte{latestBlockBytes, rf.Exchange.Reply.FinalizedBlocksHashes, rf.Addr, relaySessionHash}, nil)
```

Comment:
please introduce `sigs.Join([][]byte) []byte` helper, and use it instead of directly bytes.Join(), in all places.
motivation: this will to select the best way to implement this functionality. for example, we can do as [suggested here](https://stackoverflow.com/a/32371421) (first answer from https://stackoverflow.com/questions/32370615/what-is-the-fastest-way-to-concatenate-several-byte-together).

Reply:
done. see https://github.com/lavanet/lava/commit/c94806e8ae2de7ce31db013fbd517ae5c16de125